src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,129 @@
+                /*
+                //Send event for other systems to modify change in body temperature.
+                AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                entity.send(affectBodyTemperatureEvent);
+                deltaTemp = affectBodyTemperatureEvent.getResultValue();
+                 */
+                btc.bodyTemperature = btc.bodyTemperature + deltaTemp;
+                //Check for change in body temperature levels.
+                if ((btc.bodyTemperature < 0.3) && (btc.bodyTemperature - deltaTemp > 0.3)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.LOW));
+                }
+                if ((btc.bodyTemperature > 0.3) && (btc.bodyTemperature - deltaTemp < 0.3)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.NORMAL));
+                }
+                if ((btc.bodyTemperature > 0.6) && (btc.bodyTemperature - deltaTemp < 0.6)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.LOW));
+                }
+                if ((btc.bodyTemperature < 0.6) && (btc.bodyTemperature - deltaTemp > 0.6)) {
+                    entity.send(new BodyTemperatureChangedEvent(BodyTemperatureChangedEvent.BodyTemperatureLevel.NORMAL));
+                }
+                entity.getOwner().send(new ChatMessageEvent("Body Temperature: " + btc.bodyTemperature,
+                        entity.getOwner()));
+                entity.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, entity.getOwner()));
+                entity.saveComponent(btc);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToHigh(BodyTemperatureChangedEvent event, EntityRef player) {
+        if(event.getBodyTemperatureLevel() == BodyTemperatureChangedEvent.BodyTemperatureLevel.HIGH) {
+            player.addOrSaveComponent(new HyperthermiaComponent());
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToLow(BodyTemperatureChangedEvent event, EntityRef player) {
+        if(event.getBodyTemperatureLevel() == BodyTemperatureChangedEvent.BodyTemperatureLevel.LOW) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureChangedToNormal(BodyTemperatureChangedEvent event, EntityRef player) {
+        if(event.getBodyTemperatureLevel() == BodyTemperatureChangedEvent.BodyTemperatureLevel.NORMAL) {
+            if(player.hasComponent(HyperthermiaComponent.class)) {
+                player.removeComponent(HyperthermiaComponent.class);
+            }
+            if(player.hasComponent(HypothermiaComponent.class)) {
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
```

Comment:
These events should go into the respective Hypo-/Hyperthermia systems. 

Think about the logical units you are working with. Although this is in the same module there is no constraint that forces you to set hypo-/hyperthermia with the body temperature. In fact, we tried to decouple those from the underlying temperature system.

Ideally, those would be added in a separate PR.